src/lava/magma/runtime/runtime.py

```diff
@@ -229,25 +229,19 @@ def _run(self, run_condition):
                         data = recv_port.recv()
                         if not enum_equal(data, MGMT_RESPONSE.DONE):
                             if enum_equal(data, MGMT_RESPONSE.ERROR):
-                                # Receive all error messages from service
-                                error_msgs = []
-                                for p in self.service_to_runtime_data:
-                                    if p.probe():
-                                        num_errors = int(p.recv()[0])
-                                        for i in range(num_errors):
-                                            num_bytes = int(p.recv()[0])
-                                            data = []
-                                            for i in range(num_bytes):
-                                                data.append(int(p.recv()[0]))
-                                            error_msgs.append(
-                                                bytes(data).decode())
-
-                                # Gather all exceptions
-                                exceptions = []
-                                for msg in error_msgs:
-                                    exceptions.append(Exception(msg))
-
-                                raise Exception(exceptions)
+                                # Receive all errors from the ProcessModels
+                                error_cnt = 0
+                                for actors in \
+                                        self._messaging_infrastructure.actors:
+                                    actors.join()
+                                    if actors.exception:
+                                        _, traceback = actors.exception
+                                        print(traceback)
+                                        error_cnt += 1
+
+                                raise RuntimeError(
```

Comment:
You could define own Execptions. RuntimeError for actual errors in the Runtime and ProcessModelError for ... you know what.

Reply:
This exception is a RuntimeError, due to other Exceptions happened in the ProcessModel. The details have been printed already, this is just to stop the Runtime and tell the user that there have been other Exceptions (+ the number of them). 

For all the details the user only has to scroll upwards in the console.